packages/rs-drive-abci/src/platform_types/withdrawal/unsigned_withdrawal_txs/v0/mod.rs

```diff
@@ -48,19 +50,64 @@ impl UnsignedWithdrawalTxs {
     }
 
     /// Verifies that the collection of unsigned withdrawal transactions matches the given votes extensions
-    /// created based on these transactions
-    pub fn are_matching_with_vote_extensions(&self, other: &[VoteExtension]) -> bool {
+    /// created based on these transactions.
+    /// Returns a mapping from transactions to their corresponding vote extensions if they match, or `None` if they don't.
+    pub fn verify_and_match_with_vote_extensions<'a>(
+        &'a self,
+        other: &'a [VoteExtension],
+    ) -> Option<BTreeMap<&'a Transaction, &'a VoteExtension>> {
         if self.0.len() != other.len() {
-            return false;
-        };
+            return None;
+        }
 
-        !self.0.iter().zip(other.iter()).any(|(tx, vote_extension)| {
+        // Build a map from sign_request_id to VoteExtension
+        let mut vote_extension_map = HashMap::new();
+        for vote_extension in other {
+            // Ensure that each signature is 96 bytes (size of a bls sig)
+            if vote_extension.signature.len() != 96 {
+                return None;
+            }
+            // Ensure sign_request_id is Some
+            if let Some(sign_request_id) = &vote_extension.sign_request_id {
+                vote_extension_map.insert(sign_request_id.clone(), vote_extension);
+            } else {
+                // If sign_request_id is None, we cannot match, return None
+                return None;
+            }
+        }
+
+        let mut tx_to_vote_extension_map = BTreeMap::new();
+
+        // For each transaction, check if a matching vote extension exists
+        for tx in &self.0 {
             let extend_vote_extension = tx_to_extend_vote_extension(tx);
+            let sign_request_id = match &extend_vote_extension.sign_request_id {
+                Some(id) => id,
+                None => {
+                    // If sign_request_id is None, we cannot match, return None
+                    return None;
+                }
+            };
+
+            match vote_extension_map.get(sign_request_id) {
+                Some(vote_extension) => {
+                    if vote_extension.r#type != extend_vote_extension.r#type
+                        || vote_extension.extension != extend_vote_extension.extension
+                    {
+                        return None;
```

Comment:
it actually is logged, because on None we do ```            validation_result.add_error(AbciError::VoteExtensionMismatchReceived {
                got: commit_info.threshold_vote_extensions,
                expected: (&expected_withdrawal_transactions.clone()).into(),
            });```